catroid/src/org/catrobat/catroid/formulaeditor/SensorHandler.java

```diff
@@ -117,50 +155,92 @@ public static Double getSensorValue(Sensors sensor) {
 		switch (sensor) {
 
 			case X_ACCELERATION:
-				return (double) instance.linearAcceleartionX;
+				return (double) instance.linearAccelerationX;
 
 			case Y_ACCELERATION:
-				return (double) instance.linearAcceleartionY;
+				return (double) instance.linearAccelerationY;
 
 			case Z_ACCELERATION:
-				return (double) instance.linearAcceleartionZ;
+				return (double) instance.linearAccelerationZ;
 
 			case COMPASS_DIRECTION:
 				float[] orientations = new float[3];
-				android.hardware.SensorManager.getRotationMatrixFromVector(instance.rotationMatrix,
-						instance.rotationVector);
+				if (!instance.useRotationVectorFallback) {
```

Comment:
Hi!
If the useRotatitonVectorFallback is true - then the rotatiton-matrix is NOT obtained from the RotationVectorSensor (because this does not exist in this phone) but determined by means of the getRotationMatrix(float[],float[],float[],float[]) with accelerometer data AND macgnetic field data in line 303 in the onSensorChanged(...) method - when the MagneticField Sensor is present.
If NEITHER rotation-matrix sensor NOR magnetic field sensor are present compass direction does not work on this phone.